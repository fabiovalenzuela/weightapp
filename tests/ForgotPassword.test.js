import React from 'react';
import {TouchableOpacity} from 'react-native';
// Trouble importing shallow
import {shallow} from 'react-test-renderer/shallow';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16'
import {configure} from "enzyme";

configure({adapter: new Adapter()});

it('button renders correctly across all screens', () => {
    const tree = renderer.create(<TouchableOpacity />).toJSON();
    expect(tree).toMatchSnapshot();
})

test('test onPress functionality for Forgot Password button', () => {
    const onPressEvent = jest.passwordReset('austinmathe@gmail.com');
    onPressEvent.mockReturnValue('Switch screens invoked');

    const wrapper = shallow(<CustomLink onPress = {onPressEvent} text = 'CustomLink Component'/>);
    wrapper.find(Text).first().props.onPress();

    // Expecting the screen to change once
    expect(onPressEvent.mock.calls.length).toBe(1);
})

