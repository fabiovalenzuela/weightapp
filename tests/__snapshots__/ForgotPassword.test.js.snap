// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button renders correctly across all screens 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
/>
`;
